integrations/dell_idrac/dell_idrac.py

```diff
@@ -0,0 +1,125 @@
+            SaasConfigItem(id="password", label="Password", desc="Admin Dell iDRAC Password", type="secret", required=True),
+            SaasConfigItem(id="idrac_ip", label="iDRAC IP", desc="Admin Dell iDRAC IP", type="secret", required=True),
+            SaasConfigItem(id="user_id", label="USER_ID", desc="User ID", type="text", required=False),
+        ]
+
+    @property
+    def can_rollback(self) -> bool:
+        return True
+
+    def add_return_field(self, field: ReturnCustomField):
+        try:
+            self.return_fields.append(field)
+            Log.debug("Added return field")
+        except Exception as e:
+            raise SaasException(f"Error adding return field: {e}")
+
+    def change_password(self):
+        Log.info("Changing password for Dell iDRAC Plugin user")
+
+        admin_username = self.get_config("login")
+        admin_password = self.get_config("password")
+        idrac_ip = self.get_config("idrac_ip")
+
+        if not all([admin_username, admin_password, idrac_ip]):
+            raise SaasException('Missing required config fields: "login", "password", or "idrac_ip"')
+
+        client = DelliDRACClient(admin_username, admin_password, idrac_ip, self.user)
+        user_id = self.get_config("user_id") or client.get_user_id_from_user_fields()
+
+        if not user_id:
+            raise SaasException("Cannot determine user_id. Set it in config or user fields.")
+
+        self.__user_id = user_id
+        client.check_username_by_id(user_id)
+        self._client = client
+
+        new_password = self.user.new_password.value
+        client.change_dell_idrac_user_password(user_id, new_password)
+
+        self.add_return_field(ReturnCustomField(type="text", label="user_id", value=Secret(user_id)))
+        Log.info("Password changed successfully.")
+
+    def rollback_password(self):
+        try:
+            Log.debug("Rolling back password")
+            old_password = self.user.prior_password.value[-1]
+            self._client.change_dell_idrac_user_password(self.__user_id, old_password)
+            Log.info("Password rolled back successfully.")
+        except Exception as e:
+            raise SaasException(f"Rollback failed: {e}")
+
+
+class DelliDRACClient:
+    def __init__(self, admin_username: str, admin_password: str, idrac_ip: str, user: SaasUser):
+        self.__admin_username = admin_username
+        self.__admin_password = admin_password
+        self.__idrac_ip = idrac_ip
+        self.__user = user
+
+    def get_user_id_from_user_fields(self) -> str | None:
+        for field in self.__user.fields:
+            if field.label == "user_id" and field.values:
+                return field.values[-1]
+
+    def check_username_by_id(self, user_id: str):
+        url = ACCOUNT_SERVICE_URL.format(idrac_ip=self.__idrac_ip,user_id=user_id)
+        try:
+            response = requests.get(url, auth=(self.__admin_username, self.__admin_password))
+            if response.status_code == 200:
```

Comment:
Are we handing all the possible errors, what about 401, 403, 429 and 500? Do we want to handle, like showing specific error message. or just same message for all these failures.